lib/recursive-methods.rb

```diff
@@ -1,49 +1,107 @@
 # Authoring recursive algorithms. Add comments including time and space complexity for each method.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: ? O(n)
+# Space complexity: ? O(1)
 def factorial(n)
```

Comment:
👍 Interesting Tail recursive solution, if Ruby supported tail recursion it would be O(1) space complexity, but it doesn't so this is O(n) space complexity.